src/mono/mono/mini/interp/transform.c

```diff
@@ -4938,6 +4938,14 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 			klass = mini_get_class (method, token, generic_context);
 			CHECK_TYPELOAD (klass);
 
+			// Common in generic code:
+			// box T + unbox.any T -> nop
+			if (td->last_ins->opcode == MINT_BOX && (td->sp - 1)->klass == klass) {
+				interp_clear_ins (td, td->last_ins);
```

Comment:
You also need to update the stack type of the top of the stack. Something like `SET_TYPE (td->sp - 1, stack_type [mt], klass);`

Reply:
ah, I see, thanks!